server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -1020,17 +1022,29 @@ private SortedMap<TServerInstance,TabletServerStatus> gatherTableInformation(
             badServers.remove(server);
           }
         }
-      });
-    }
-    tp.shutdown();
-    try {
-      tp.awaitTermination(Math.max(10000, rpcTimeout / 3), MILLISECONDS);
-    } catch (InterruptedException e) {
-      log.debug("Interrupted while fetching status");
+      }));
+    }
+    final long timeToWaitForCompletion = Math.max(10000, rpcTimeout / 3);
+    long currTime = NANOSECONDS.toMillis(System.nanoTime());
+    final long timeToCancelTasks = currTime + timeToWaitForCompletion;
+    // Wait for all tasks to complete
+    while (!tasks.isEmpty()) {
+      boolean cancel = (currTime > timeToCancelTasks);
```

Comment:
The time returned by `System.nanoTime()` is only valid when compared relative to another nano time.  The value is relative to some arbitrary counter in the JVM and can be negative.

This should be sometime like ```
boolean cancel = ((currTime - System.nanoTime()) > MILLISECONDS.toNanos(timeToCancelTasks));
``` 

If may also help if all times were kept as nanos - or converted to nanos for the calculations and only converting back to millis (or seconds) for display to the user.  Using one set of units consistently may help limit conversion errors if a conversion was missed.

Reply:
Thx @EdColeman for the explanation. I believe I have resolved this in 860fe79.